cmd/bots/internal/environment.go

```diff
@@ -517,6 +380,98 @@ func (tgEnv *TelegramBotEnvironment) IsAlreadySubscribed(alertType entities.Aler
 	return ok
 }
 
+func ScheduleNodesStatus(
+	taskScheduler chrono.TaskScheduler,
+	requestType chan<- pair.RequestPair,
+	responsePairType <-chan pair.ResponsePair, bots []messaging.Bot) {
+
+	_, err := taskScheduler.ScheduleWithCron(func(ctx context.Context) {
+		urls, err := RequestNodesList(requestType, responsePairType, false)
+		if err != nil {
+			log.Printf("failed to request list of nodes, %v", err)
+		}
+		additionalUrls, err := RequestNodesList(requestType, responsePairType, true)
+		if err != nil {
+			log.Printf("failed to request list of additional nodes, %v", err)
+		}
+		urls = append(urls, additionalUrls...)
+
+		nodesStatus, err := RequestNodesStatus(requestType, responsePairType, urls)
+		if err != nil {
+			log.Printf("failed to request status of nodes, %v", err)
+		}
+
+		for _, bot := range bots {
+
+			var handledNodesStatus string
+			statusCondition := StatusCondition{AllNodesAreOk: false, NodesNumber: 0, Height: ""}
+			switch bot.(type) {
+			case *TelegramBotEnvironment:
+				handledNodesStatus, statusCondition, err = HandleNodesStatus(nodesStatus, Html)
+				if err != nil {
+					log.Printf("failed to handle status of nodes, %v", err)
+				}
+			case *DiscordBotEnvironment:
+				handledNodesStatus, statusCondition, err = HandleNodesStatus(nodesStatus, Markdown)
+				if err != nil {
+					log.Printf("failed to handle status of nodes, %v", err)
+				}
+			default:
+				log.Println("failed to schedule nodes status, unknown bot type")
+				return
+			}
+
+			if statusCondition.AllNodesAreOk {
+				var msg string
+				switch bot.(type) {
+				case *TelegramBotEnvironment:
+					msg = fmt.Sprintf("Status %s\n\nAll <b>%d</b> nodes have the same hashes on height <code>%s</code>", messages.TimerMsg, statusCondition.NodesNumber, statusCondition.Height)
+				case *DiscordBotEnvironment:
+					msg = fmt.Sprintf("Status %s\n\nAll %d nodes have the same hashes on height %s", messages.TimerMsg, statusCondition.NodesNumber, statusCondition.Height)
+					msg = fmt.Sprintf("```yaml\n%s\n```", msg)
+				default:
+					log.Println("failed to schedule nodes status, unknown bot type")
+					return
+				}
+				bot.SendMessage(msg)
+				continue
+			}
+			var msg string
+			switch bot.(type) {
+			case *TelegramBotEnvironment:
+				msg = fmt.Sprintf("Status %s\n\n%s", messages.TimerMsg, handledNodesStatus)
+			case *DiscordBotEnvironment:
+				msg = fmt.Sprintf("```yaml\nStatus %s\n\n%s\n```", messages.TimerMsg, handledNodesStatus)
+			}
+			bot.SendMessage(msg)
+		}
+
+	}, scheduledTimeExpression)
+
+	if err != nil {
+		taskScheduler.Shutdown()
```

Comment:
Return error `if err != nil`. Closing `taskScheduler` in case of error is caller's responsibility.